rpc/backend/backend.go

```diff
@@ -43,155 +42,126 @@ var _ Backend = (*EthermintBackend)(nil)
 
 // EthermintBackend implements the Backend interface
 type EthermintBackend struct {
-	ctx       context.Context
-	clientCtx clientcontext.CLIContext
-	logger    log.Logger
-	gasLimit  int64
+	ctx         context.Context
+	clientCtx   client.Context
+	queryClient *rpctypes.QueryClient // gRPC query client
+	logger      log.Logger
+	gasLimit    int64
```

Comment:
does this describe the block gas limit?

Reply:
idk why is the gas limit a param here. Will update